SimplenoteUITests/AllNotesClass.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  AllNotesClass.swift
+//  SimplenoteUITests
+//
+//  Created by Sergiy Fedosov on 03.02.2021.
+//  Copyright © 2021 Automattic. All rights reserved.
+//
+
+import XCTest
+
+class AllNotes {
+
+    class func openNote(noteName: String) {
+        app.tables.cells[noteName].tap()
+    }
+    
+    class func isOpen() -> Bool {
+        return app.navigationBars[uidNavBar_AllNotes].exists
+    }
+
+    class func open() {
+        if !isOpen() {
+            app.navigationBars.element.buttons[uidButton_Menu].tap()
+            app.tables.staticTexts[uidButton_AllNotes].tap()
+        }
+    }
+
+    class func addNoteTap() {
+        app.navigationBars[uidNavBar_AllNotes].buttons[uidButton_NewNote].tap()
+    }
+
+    class func createNoteAndLeaveEditor (noteName: String) {
+        AllNotes.addNoteTap()
+        NoteEditor.clearAndEnterText(enteredValue: noteName)
+        NoteEditor.leaveEditor()
+    }
+
+    class func trashNote(noteName: String) {
+        Table.trashNote(noteName: noteName)
+    }
+
+    class func getNotesNumber() -> Int {
+        return Table.getNotesNumber()
+    }
+
+    class func clearAllNotes() {
+        let notesNumber = AllNotes.getNotesNumber()
+        let cellsNum = app.tables.element.children(matching: .cell).count
+        var startingIndex: Int
+
+        if notesNumber == cellsNum {
+            // Depending on what happened before, the cells numbering
+            // might not include "All Notes", "Trash" and "Settings" cells...
+            startingIndex = 0
+        } else {
+            // Or might include them
+            startingIndex = 3
+        }
+
+        for _ in 0..<notesNumber {
+            let cell = app.tables.cells.element(boundBy: startingIndex)
+            cell.swipeLeft()
+            cell.buttons[uidButton_NoteCell_Trash].tap()
+        }
+    }
+
+    class func waitForLoad() {
+        let allNotesNavBar = app.navigationBars[uidNavBar_AllNotes]
+        sleep(4)
+        while allNotesNavBar.staticTexts[uidText_AllNotes_InProgress].exists {
+        }
+    }
+}
+
+class AllNotesAssert {
+
+    class func noteExists(noteName: String) {
+        XCTAssertTrue(app.tables.cells[noteName].exists, "\"" + noteName + noteNotFoundInAllNotes)
+    }
```

Comment:
Out of curiosity, why did you decide to split the assertions in a dedicated class as opposed to keep them in `AllNotes` and prefix the function name with "assert", e.g.:

```swift
AllNotes.assertNoteExists(noteName: ...)
```

Just out of curiosity though, I don't see any issue in this.

Reply:
I think this happened because I assumed that a class with actions and assertion can grow enough that it should be split into separate files, and also because it's easier to have an overview of what's included into the class, like this:

<img width="261" alt="Screenshot 2021-02-15 at 14 26 11" src="https://user-images.githubusercontent.com/73365754/107946506-ccb49c00-6f99-11eb-83d4-9ec927ad183f.png">

I agree that it makes a very little difference, and just creates another class. I'd tend to keep it this way meanwhile, in order not to overload the review with even more changes.